packages/ember-glimmer/lib/utils/process-args.js

```diff
@@ -51,7 +51,9 @@ class SimpleArgs {
       let ref = namedArgs.get(name);
       let value = attrs[name];
 
-      if (ref[UPDATE]) {
+      if (typeof value === 'function') {
+        attrs[name] = value;
```

Comment:
This would prevent passing a class down to a child component.  Given the following setup (we should add a test for this):

```hbs
{{foo-bar klass=ModelClass}}
```

```js
{{! parent component }}
export default Ember.Component.extend({
  init() {
    this._super();
    this.klass = Ember.Object.extend();
  }
});
```

```js
// foo-bar component

export default Ember.Component.extend({
  actions: {
    updateKlass(newKlass) {
      this.attrs.klass.update(newKlass);
    }
  }
});
```